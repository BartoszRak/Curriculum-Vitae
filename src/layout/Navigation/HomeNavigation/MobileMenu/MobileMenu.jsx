import React, { useState } from 'react'
import PropTypes from 'prop-types'
import {
  IconButton,
  Modal,
  MenuList,
  MenuItem,
} from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'

import { useScrollTo } from '~hooks'

import useStyle from './MobileMenu.style'

export function MobileMenu({ classes: overridingClasses, className, routes }) {
  const classes = { ...useStyle(), ...overridingClasses }
  const [toggle, setToggle] = useState(false)
  const { scrollTo } = useScrollTo()
  return (
    <div className={className}>
      <IconButton
        data-testid="mobile-menu-open-icon"
        onClick={() => {
          setToggle(!toggle)
        }}
      >
        <MenuIcon color="secondary" />
      </IconButton>
      <Modal
        BackdropProps={{
          classes: {
            root: classes.backdrop,
          }
        }}
        className={classes.dialog}
        data-testid="mobile-menu-modal"
        disableAutoFocus
        onBackdropClick={() => setToggle(!toggle)}
        open={toggle}
      >
        <MenuList className={classes.list}>
          {routes.map(route => (
            <MenuItem
              className={classes.item}
              key={route.name}
              onClick={() => {
                setToggle(!toggle)
                scrollTo(route.elementId)
              }}
            >
              {route.name}
            </MenuItem>
          ))}
        </MenuList>
      </Modal>
    </div>
  )
}

MobileMenu.propTypes = {
  classes: PropTypes.object,
  className: PropTypes.string,
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      elementId: PropTypes.string,
      name: PropTypes.string,
    })
  ).isRequired,
}

export default MobileMenu
