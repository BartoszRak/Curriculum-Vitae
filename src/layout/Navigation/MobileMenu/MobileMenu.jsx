import React, { useState } from 'react'
import PropTypes from 'prop-types'
import {
  withStyles,
  IconButton,
  Dialog,
  MenuList,
  MenuItem,
} from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'

import { useScrollTo } from '~hooks'

import style from './MobileMenu.style'

export function MobileMenu({ classes, routes }) {
  const [toggle, setToggle] = useState(false)
  const { scrollTo } = useScrollTo()
  return (
    <React.Fragment>
      <IconButton
        onClick={() => {
          setToggle(!toggle)
        }}
      >
        <MenuIcon color="secondary" />
      </IconButton>
      <Dialog
        className={classes.dialog}
        classes={{
          modal: classes.dialogModal,
          paper: classes.dialogPaper,
        }}
        onBackdropClick={() => setToggle(!toggle)}
        open={toggle}
      >
        <MenuList className={classes.list}>
          {routes.map(route => (
            <MenuItem
              classes={{
                label: classes.buttonLabel,
                root: classes.button,
              }}
              color="secondary"
              key={route.name}
              onClick={() => {
                setToggle(!toggle)
                scrollTo(route.elementId)
              }}
            >
              {route.name}
            </MenuItem>
          ))}
        </MenuList>
      </Dialog>
    </React.Fragment>
  )
}

MobileMenu.propTypes = {
  classes: PropTypes.object,
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      elementId: PropTypes.string,
      name: PropTypes.string,
    })
  ).isRequired,
}

export default withStyles(style)(MobileMenu)
