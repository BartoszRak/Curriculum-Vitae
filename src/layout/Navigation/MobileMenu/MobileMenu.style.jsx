import backgroundUrl from '~assets/images/landscape-mountains-nature-dark-blur.jpg'

export default ({ palette }) => ({
  backdrop: {
    background: `url(${backgroundUrl})`,
    backgroundAttachment: 'fixed',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
  },
  dialog: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'center',
  },
  item: {
    color: palette.secondary.main,
    fontSize: '1.3rem',
  },
  list: {
    textShadow: '0 0 5px rgba(0, 0, 0, 0.4)',
  },
})
