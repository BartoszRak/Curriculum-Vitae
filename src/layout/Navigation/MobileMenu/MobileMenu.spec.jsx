import React from 'react'
import { shallow } from 'enzyme'

import { MobileMenu } from './MobileMenu'

const initComponent = overrides => {
  const mockProps = {
    classes: {},
    routes: [
      {
        elementId: 'testElementId1',
        name: 'testElementName1',
      },
      {
        elementId: 'testElementId2',
        name: 'testElementName2',
      },
    ],
  }
  const mockMethods = {}
  const wrapper = shallow(<MobileMenu {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('global: MobileMenu', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
