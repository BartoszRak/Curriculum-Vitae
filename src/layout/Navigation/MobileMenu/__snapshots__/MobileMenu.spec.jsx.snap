// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: MobileMenu should render as expected 1`] = `
<Fragment>
  <IconButtonMock
    onClick={[Function]}
  >
    <pure(MenuIcon)
      color="secondary"
    />
  </IconButtonMock>
  <DialogMock
    classes={
      Object {
        "modal": undefined,
        "paper": undefined,
      }
    }
    onBackdropClick={[Function]}
    open={false}
  >
    <MenuListMock>
      <MenuItemMock
        classes={
          Object {
            "label": undefined,
            "root": undefined,
          }
        }
        color="secondary"
        key="testElementName1"
        onClick={[Function]}
      >
        testElementName1
      </MenuItemMock>
      <MenuItemMock
        classes={
          Object {
            "label": undefined,
            "root": undefined,
          }
        }
        color="secondary"
        key="testElementName2"
        onClick={[Function]}
      >
        testElementName2
      </MenuItemMock>
    </MenuListMock>
  </DialogMock>
</Fragment>
`;
